v2/gcs-to-splunk/src/main/java/com/google/cloud/teleport/v2/templates/GCSToSplunk.java

```diff
@@ -0,0 +1,253 @@
+
+  /**
+   * The {@link GCSToSplunkOptions} class provides the custom execution options passed by the
+   * executor at the command-line.
+   */
+  public interface GCSToSplunkOptions extends CsvConverters.CsvPipelineOptions, SplunkOptions {
+
+    @Description("Pattern of where to write errors, ex: gs://mybucket/somepath/errors.txt")
+    ValueProvider<String> getInvalidOutputPath();
+
+    void setInvalidOutputPath(ValueProvider<String> value);
+  }
+
+  /**
+   * Main entry-point for the pipeline. Reads in the command-line arguments, parses them, and
+   * executes the pipeline.
+   *
+   * @param args Arguments passed in from the command-line.
+   */
+  public static void main(String[] args) {
+    GCSToSplunkOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(GCSToSplunkOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs the pipeline to completion with the specified options. This method does not wait until the
+   * pipeline is finished before returning. Invoke {@code result.waitUntilFinish()} on the result
+   * object to block until the pipeline is finished running if blocking programmatic execution is
+   * required.
+   *
+   * @param options The execution options.
+   * @return The pipeline result.
+   */
+  public static PipelineResult run(GCSToSplunkOptions options) {
+    Pipeline pipeline = Pipeline.create(options);
+
+    CoderRegistry registry = pipeline.getCoderRegistry();
+    registry.registerCoderForClass(SplunkEvent.class, SplunkEventCoder.of());
+    registry.registerCoderForType(
+        FAILSAFE_ELEMENT_CODER.getEncodedTypeDescriptor(), FAILSAFE_ELEMENT_CODER);
+
+    /*
+     * Steps:
+     *  1) Step 1: Read CSV file(s) from Cloud Storage using {@link CsvConverters.ReadCsv}.
+     *  2) Apply user provided UDF (if any) on the input strings.
+     *  3) Convert successfully transformed messages into SplunkEvent objects
+     *  4) Write SplunkEvents to Splunk's HEC end point.
+     *  4a) Wrap write failures into a FailsafeElement.
+     *  5) Collect errors from UDF transform (#3), SplunkEvent transform (#4)
+     *     and writing to Splunk HEC (#5) and place into a GCS folder.
+     */
+
+    PCollectionTuple convertToEventTuple =
+        pipeline
+            // 1) Read CSV file(s) from Cloud Storage using {@link CsvConverters.ReadCsv}.
+            .apply(
+                "ReadCsv",
+                CsvConverters.ReadCsv.newBuilder()
+                    .setCsvFormat(options.getCsvFormat())
+                    .setDelimiter(options.getDelimiter())
+                    .setHasHeaders(options.getContainsHeaders())
+                    .setInputFileSpec(options.getInputFileSpec())
+                    .setHeaderTag(CSV_HEADERS)
+                    .setLineTag(CSV_LINES)
+                    .setFileEncoding(options.getCsvFileEncoding())
+                    .build())
+
+            // 2) Apply user provided UDF (if any) on the input strings.
+            .apply(
```

Comment:
If the user doesn't provide a UDF, then this step will still be part of the graph. That's confusing and wasteful. It's unavoidable with Classic Templates, but for Flex Templates, we should take advantage of the flexibility with the graph when it makes sense, so I'd only add this node when necessary.

An example template already adding UDFs only when necessary is [Pub/Sub Proto to BigQuery](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/pubsub-binary-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/PubsubProtoToBigQuery.java).